crates/viewer/re_space_view_spatial/src/visualizers/assets3d.rs

```diff
@@ -144,38 +146,61 @@ impl VisualizerSystem for Asset3DVisualizer {
 
         let mut instances = Vec::new();
 
-        super::entity_iterator::process_archetype::<Self, Asset3D, _>(
+        super::entity_iterator::process_archetype2::<Self, Asset3D, _>(
             ctx,
             view_query,
             context_systems,
             |ctx, spatial_ctx, results| {
-                use re_space_view::RangeResultsExt as _;
-
-                let resolver = ctx.recording().resolver();
+                use re_space_view::RangeResultsExt2 as _;
 
-                let blobs = match results.get_required_component_dense::<Blob>(resolver) {
-                    Some(blobs) => blobs?,
-                    _ => return Ok(()),
+                let Some(all_blob_chunks) = results.get_required_chunks(&Blob::name()) else {
+                    return Ok(());
                 };
 
-                let media_types = results.get_or_empty_dense(resolver)?;
-                let transforms = results.get_or_empty_dense(resolver)?;
+                let timeline = ctx.query.timeline();
+                let all_blobs_indexed = all_blob_chunks.iter().flat_map(|chunk| {
+                    itertools::izip!(
+                        chunk.iter_component_indices(&timeline, &Blob::name()),
+                        chunk.iter_buffer::<u8>(&Blob::name())
+                    )
+                });
+                let all_media_types = results.iter_as(timeline, MediaType::name());
+
+                // TODO(#6831): we have to deserialize here because `OutOfTreeTransform3D` is
+                // still a complex type at this point.
```

Comment:
fixed on a PR near you :smile:

Reply:
![image](https://github.com/user-attachments/assets/094a188f-3eec-459e-8137-60c2fd67fa2d)